pkg/secretstorage/awsstorage/aws.go

```diff
@@ -90,40 +99,54 @@ func (s *AwsSecretStorage) Store(ctx context.Context, id secretstorage.SecretID,
 }
 
 func (s *AwsSecretStorage) Get(ctx context.Context, id secretstorage.SecretID) ([]byte, error) {
-	dbgLog := lg(ctx).V(logs.DebugLevel).WithValues("secretID", id)
+	return s.doGetWithRetry(ctx, id, secretReadRetryCount)
+}
 
+func (s *AwsSecretStorage) doGetWithRetry(ctx context.Context, id secretstorage.SecretID, retryTries uint64) ([]byte, error) {
```

Comment:
I would prefer that we refactor this method a little bit and maybe split it into separate functions. It would help with testability and readability since following the logic 5 levels deep is challenging for me.

If that's not possible, maybe more comments would help.

Reply:
I returned most of the implenentation back to the original method and left separate only stuff needed for repeat.

Reply:
Ans some more comments added